### PR TITLE
GPU Info

### PR DESCRIPTION
**Pull request by 907Resident**
# TL;DR

Overall, I wanted an argument in `watermark` to allow the user to view GPU information. 

# Details

- I stumbled upon this issue #31 while trying to view my GPU capabilities 
- Since the issue has been open since 2017, I have taken a stab by submitting a PR to the argument `gpu` to `watermark`
- Note: This will only work with GPUs, where NVIDIA drivers are installed already. In the issue, @dartdog mentions, and I agree with, most people are not using AMD GPUs for data science. Therefore, I think that only accounting for NVIDIA drivers should be fine
- Lastly, I added a new file in `tests/` called `test_gpu_info.py`, which should allow the maintainers of this repo to quickly run a `pytest` to ensure that the `gpu` argument works as the maintainers see fit

### CODE REVIEW

**Comment by rasbt** [comment]
Awesome! This is a very exciting PR! Thanks!

**Comment by rasbt** [comment]
I just tried it on a machine with NVIDIA GPUs, and it couldn't find these. How did it look on your machine, am I missing any requirements @907Resident ?
Screenshot 2023-05-22 at 2 06 05 PM

**Comment by 907Resident** [comment]
Hi @rasbt,

I think I accidentally provided the incorrect pythonic interface for the gpu. As I mentioned in commit [87e7201](https://github.com/rasbt/watermark/pull/90/commits/87e7201c909de03ef9cf93ab0bd75f16a797af7e), the correct package is `py3nvml`. 

I have made the correction, and the change can be seen in [this Colab notebook](https://colab.research.google.com/drive/1ewLCMss5HcK-H2kCj3Ra4pfn-drsYe0z?usp=sharing) (assuming the user has Colab Pro with GPU).  The output can be viewed below.

image

**Comment by 907Resident** [comment]
I also noticed that as of commit [ab5cf05](https://github.com/rasbt/watermark/pull/90/commits/ab5cf052425099b27fc1355ca9d15a805f2773d8), that the automated checks have failed. But my local execution of `pytest` appears to be working:

image

But for some reason that I do no understand, the automated CI process is unable to install `py3nvml`. 

image

Nonetheless, I think once `py3nvml` is installed, I believe that you will be able to get the name of the GPU being used with my additions. If there is more functionality that you want to see, just let me know.

**Comment by rasbt** [comment]
Thanks for the update! I did some fixes in the formatting and the CI works now! I'll also try it on a non-Colab GPU machine to make sure.

**Comment by rasbt** [comment]
I changed it to always use the indentation now. In hindsight, I think this should look better since it avoids the double-colon.

I.e., it's now

```
GPU Info:
  GPU 0: NVIDIA A100-SXM4-40GB
  GPU 1: NVIDIA A100-SXM4-40GB
  GPU 2: NVIDIA A100-SXM4-40GB
  GPU 3: NVIDIA A100-SXM4-40GB
  GPU 4: NVIDIA A100-SXM4-40GB
  GPU 5: NVIDIA A100-SXM4-40GB
  GPU 6: NVIDIA A100-SXM4-40GB
  GPU 7: NVIDIA A100-SXM4-40GB
```

or 

```
GPU Info:
  GPU 0: NVIDIA A100-SXM4-40GB

```

**Comment by 907Resident** [comment]
Thanks for the updates and integrating my changes. 




> I changed it to always use the indentation now. In hindsight, I think this should look better since it avoids the double-colon.
> 
> I.e., it's now
> 
> ```
> GPU Info:
>   GPU 0: NVIDIA A100-SXM4-40GB
>   GPU 1: NVIDIA A100-SXM4-40GB
>   GPU 2: NVIDIA A100-SXM4-40GB
>   GPU 3: NVIDIA A100-SXM4-40GB
>   GPU 4: NVIDIA A100-SXM4-40GB
>   GPU 5: NVIDIA A100-SXM4-40GB
>   GPU 6: NVIDIA A100-SXM4-40GB
>   GPU 7: NVIDIA A100-SXM4-40GB
> ```
> 
> or
> 
> ```
> GPU Info:
>   GPU 0: NVIDIA A100-SXM4-40GB
> ```

Yes, I agree. I only have access to a machine with one GPU; so, I did not think about formatting for multiple GPUs. But your formatting looks much better.

@rasbt, at this point what do we need to do to merge this PR? Anything else you need me to add?

**Comment by rasbt** [comment]
@907Resident It looks all good to me know! Just added a changelog entry and am happy to merge (and make version release then)! Thanks so much for the PR. It's a super nice contribution!
